calico_kubernetes/calico_kubernetes_cni.py

```diff
@@ -103,30 +131,29 @@ def delete(args):
                       container_id=container_id)
 
     # Delete profile if only member
-    if datastore_client.profile_exists(net_name) and \
-       len(datastore_client.get_profile_members(net_name)) < 1:
+    if _datastore_client.profile_exists(profile_name) and \
```

Comment:
This code isn't safe -- what happens if someone creates a pod somewhere else in the cluster that uses this profile?